Backpack/Calendar/Classes/BPKCalendar.m

```diff
@@ -146,39 +152,54 @@ - (instancetype)initWithFrame:(CGRect)frame {
 - (instancetype)initWithMinDate:(BPKSimpleDate *)minDate
                         maxDate:(BPKSimpleDate *)maxDate
                   configuration:(BPKCalendarConfiguration *)configuration
-         selectionConfiguration:(BPKCalendarSelectionConfiguration *)selectionConfiguration {
-    _configuration = configuration;
-
-    return [self initWithMinDate:minDate maxDate:maxDate selectionConfiguration:selectionConfiguration];
-}
-
-- (instancetype)initWithMinDate:(BPKSimpleDate *)minDate
-                        maxDate:(BPKSimpleDate *)maxDate
-         selectionConfiguration:(BPKCalendarSelectionConfiguration *)selectionConfiguration {
+         selectionConfiguration:(BPKCalendarSelectionConfiguration *)selectionConfiguration
+                       calendar:(NSCalendar *)calendar
+                   dateProvider:(id<BPKMonthDateProvider>)dateProvider {
     BPKAssertMainThread();
-    if (_configuration == nil) {
-        _configuration = [BPKCalendarTrafficLightConfiguration new];
-    }
-
     self = [super initWithFrame:CGRectZero];
 
     if (self) {
-        self.selectionConfiguration = selectionConfiguration;
         self.minDate = minDate;
         self.maxDate = maxDate;
+        _configuration = configuration != nil ? configuration : [BPKCalendarTrafficLightConfiguration new];
+        self.selectionConfiguration = selectionConfiguration;
+        _gregorian = calendar;
```

Comment:
It is probably misleading to assign an NSCalendar to the _gregorian variable. Is there a specific reason this new init requires the end-user to pass in a Calendar? 

Would it be ok for us to init a gregorian calendar like the original initializer?